lib/ravelin/event.rb

```diff
@@ -73,9 +73,6 @@ def validate_top_level_payload_params
           :payment_method_id, :payment_method
         )
         validate_payload_inclusion_of :order_id
-      when :checkout
-        validate_payload_inclusion_of :customer, :order,
-          :payment_method, :transaction
```

Comment:
⚠️  Caution: If these fields ever go missing for the main checkout for any reason, it could have an affect on our fraud detection capabilities (anywhere from broken to degraded).

How else would we notice if they went missing? Is there a way we can detect the kind of payment being made and validate accordingly?

I will see if I can find out from someone closer to this what the risks are.

Reply:
This doesn't match their spec though - you can see clearly in the docs which fields are required and which ones are important. I would posit that we should be adding these sort of checks at the callsite. This gem should adhere to their API spec IMO

Reply:
Yes my mistake, I assumed this was in orderweb. I'll be quiet :)